### ISSUE TITLE
IonTokenTroubleshootingWindow crashes

### ISSUE THREAD

**Issue by kring** [CesiumGS/cesium-unity#207]
On any tileset, click the "Troubleshoot Token" button on the Inspector panel. The troubleshooting panel appears, but then Unity crashes with this strack trace:

```
Obtained 31 stack frames
0x00007ffaa27bdfa8 (CesiumForUnityNative-Editor) [traits.h:109] async::detail::invoke_fake_void<<lambda_1eaefc7302ab98935558272345ed73d0>,CesiumIonClient::Response<CesiumIonClient::Profile> > 
0x00007ffaa27bae0b (CesiumForUnityNative-Editor) [task_base.h:398] async::detail::task_func<CesiumAsync::CesiumImpl::ImmediateScheduler<CesiumAsync::CesiumImpl::QueuedScheduler>,async::detail::continuation_exec_func<CesiumAsync::CesiumImpl::ImmediateScheduler<CesiumAsync::CesiumImpl::QueuedScheduler>,async::task<CesiumIonClient::Response<CesiumIonClient::Profile> >,async::detail::fake_void,<lambda_1eaefc7302ab98935558272345ed73d0>,std::integral_constant<bool,1>,0>,async::detail::fake_void>::run 
0x00007ffaa27a04be (CesiumForUnityNative-Editor) [scheduler.h:165] async::detail::schedule_task<CesiumAsync::CesiumImpl::ImmediateScheduler<CesiumAsync::CesiumImpl::QueuedScheduler> > 
0x00007ffaa27a03c8 (CesiumForUnityNative-Editor) [task_base.h:101] async::detail::task_base::run_continuation<CesiumAsync::CesiumImpl::ImmediateScheduler<CesiumAsync::CesiumImpl::QueuedScheduler> > 
0x00007ffaa27baf8c (CesiumForUnityNative-Editor) [task_base.h:417] async::detail::task_func<CesiumAsync::CesiumImpl::ImmediateScheduler<CesiumAsync::CesiumImpl::QueuedScheduler>,async::detail::continuation_exec_func<CesiumAsync::CesiumImpl::ImmediateScheduler<CesiumAsync::CesiumImpl::QueuedScheduler>,async::task<CesiumIonClient::Response<CesiumIonClient::Profile> >,async::detail::fake_void,<lambda_1eaefc7302ab98935558272345ed73d0>,std::integral_constant<bool,1>,0>,async::detail::fake_void>::schedule 
0x00007ffaa27a467f (CesiumForUnityNative-Editor) [task_base.h:115] <lambda_748db4bbc179da6db048344469bf4b84>::operator() 
0x00007ffaa279fde8 (CesiumForUnityNative-Editor) [continuation_vector.h:221] async::detail::continuation_vector::flush_and_lock<<lambda_748db4bbc179da6db048344469bf4b84> > 
0x00007ffaa27dd693 (CesiumForUnityNative-Editor) [task_base.h:398] async::detail::task_func<CesiumAsync::CesiumImpl::ImmediateScheduler<CesiumAsync::CesiumImpl::QueuedScheduler>,async::detail::continuation_exec_func<CesiumAsync::CesiumImpl::ImmediateScheduler<CesiumAsync::CesiumImpl::QueuedScheduler>,async::task<std::shared_ptr<CesiumAsync::IAssetRequest> >,CesiumIonClient::Response<CesiumIonClient::Profile>,<lambda_5dedf0dce7adcf57697cfb093d21f5eb>,std::integral_constant<bool,1>,0>,CesiumIonClient::Response<CesiumIonClient::Profile> >::run 
0x00007ffaa28315ab (CesiumForUnityNative-Editor) [scheduler.cpp:215] async::fifo_scheduler::try_run_one_task 
0x00007ffaa2831208 (CesiumForUnityNative-Editor) [scheduler.cpp:222] async::fifo_scheduler::run_all_tasks 
0x00007ffaa282eabe (CesiumForUnityNative-Editor) [QueuedScheduler.cpp:11] CesiumAsync::CesiumImpl::QueuedScheduler::dispatchQueuedContinuations 
0x00007ffaa27ab3da (CesiumForUnityNative-Editor) [CesiumIonSession.cpp:252] DotNet_CesiumForUnity_CesiumIonSession_Tick 
0x00000240729b1324 (Mono JIT Code) (wrapper managed-to-native) CesiumForUnity.CesiumIonSession:DotNet_CesiumForUnity_CesiumIonSession_Tick (intptr,CesiumForUnity.CesiumIonSession/ImplementationHandle)
0x00000240729b11d3 (Mono JIT Code) [CesiumIonSession-generated.cs:199] CesiumForUnity.CesiumIonSession:Tick () 
0x0000024071971883 (Mono JIT Code) [CesiumEditorUtility.cs:90] CesiumForUnity.CesiumEditorUtility:UpdateIonSession () 
0x0000024071925630 (Mono JIT Code) UnityEditor.EditorApplication:Internal_CallUpdateFunctions ()
0x0000023f3d4283ed (Mono JIT Code) (wrapper runtime-invoke) object:runtime_invoke_void (object,intptr,intptr,intptr)
0x00007ffa885cfeb4 (mono-2.0-bdwgc) [mini-runtime.c:3445] mono_jit_runtime_invoke 
0x00007ffa8850e764 (mono-2.0-bdwgc) [object.c:3066] do_runtime_invoke 
0x00007ffa8850e8fc (mono-2.0-bdwgc) [object.c:3113] mono_runtime_invoke 
0x00007ff67befcc34 (Unity) scripting_method_invoke
0x00007ff67bedc4d4 (Unity) ScriptingInvocation::Invoke
0x00007ff67bed70e5 (Unity) ScriptingInvocation::Invoke<void>
0x00007ff67c01693b (Unity) Scripting::UnityEditor::EditorApplicationProxy::Internal_CallUpdateFunctions
0x00007ff67c961c92 (Unity) SceneTracker::Update
0x00007ff67cb13f26 (Unity) Application::TickTimer
0x00007ff67cf68e5a (Unity) MainMessageLoop
0x00007ff67cf6d72b (Unity) WinMain
0x00007ff67e32224e (Unity) __scrt_common_main_seh
0x00007ffaea0726bd (KERNEL32) BaseThreadInitThunk
0x00007ffaebe4dfb8 (ntdll) RtlUserThreadStart
```

This happens to be a `RelWithDebInfo` build, but I doubt it matters.

There's a good chance this broke with recent Reinterop or ImplementationHandle changes.

**Comment by j9liu** [comment]
@kring do you have an idea of which commit / PR broke this? I merged main with a branch I'm working on but I'm not seeing this crash.

**Comment by kring** [comment]
@j9liu I see it crash consistently in main. More details:

* Unity 2021.3.17f1
* Both Debug and RelWithDebInfo configurations of the native code crash similarly.
* I cleaned and published Reinterop, then modified a source file to make sure it re-ran.
* I did `git clean -d -f -x` in both cesium-unreal-samples and in the Packages/com.cesium.unity directories, and then rebuilt everything.

But if it's not happening for you, I'll investigate.
